### PR TITLE
Create register and login functionality, create User schema 

### PR DESCRIPTION
**Pull request by Jennifercheukyin**
**Status**: ready for review

**Done in this PR:** 
- Register users 
- Log in users 
- User schema creation 

**TODO:** 
- [x] Test scenarios 
- [ ] Unit test both register and login functions (I'll leave this to another PR)

### CODE REVIEW

File: front-end/src/components/LoginForm.js [hunk 1]
```diff
@@ -1,47 +1,88 @@
...
+              />
             </div>
             <br />
             <div>
               <label htmlFor="password">
                 <b>Password:</b>
               </label>
-              <input type="text" id="password" required />
+              <input
+                type="text"
```
Comment (azc242) [inline front-end/src/components/LoginForm.js:68]: Small nit (forgot to point this out previously), but can you change this (and in registration) to `input type="password"`?

File: back-end/app.js [hunk 2]
```diff
@@ -87,6 +91,68 @@ const plaidClient = new PlaidApi(configuration);
 app.use(express.json()); // decode JSON-formatted incoming POST data
 app.use(express.urlencoded({ extended: true })); // decode url-encoded incoming POST data
 
+// Get user info from frontend and sign 
+app.post('/api/login', async (req, res) => {
```
Comment (azc242) [inline back-end/app.js:95]: On my browser, login doesn't go through. The request to backend never returns a response, likely due to the error (see screenshot). (Also I don't know why my screenshot turned out so red, please ignore)
image

**Review by azc242 — CHANGES_REQUESTED** [review]
The login doesn't backend doesn't seem to go through on my machine (see screenshot in comments). Maybe I am doing something wrong. I did `npm i` on both front and back end.

File: back-end/app.js (hunk 2 above)
Comment (Jennifercheukyin) [reply back-end/app.js:95]: The error is due to the lack of secret key for jwt, which I put it in .env file and forgot to mention it. Just sent it to you on Slack

Comment (azc242) [reply back-end/app.js:95]: added secret key to `.env` fixed it. If you are having issues with this, please DM Jennifer or me for the secret key (I won't post it here, obviously)

**Review by azc242 — APPROVED** [review]
Just the small nit on passwords, but otherwise looks good! Thanks for handling the login and auth! 🚢 💯

**Review by azc242 — APPROVED** [review]
LGTM, thanks for updating the password inputs!

**Pull request merged**